### PR TITLE
Added android functionality

### PR DESCRIPTION
**Pull request by rohanhc**

### CODE REVIEW

File: android/KochavaPlugin.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+	public void onCreate(Activity activity, Bundle savedInstanceState) {
+		this.mActivity = activity;
+
+		logger.log("ROHAN Kochava onCreate");
+		PackageManager manager = activity.getPackageManager();
+		String kochavaKey = "";
+		try {
+			Bundle meta = manager.getApplicationInfo(activity.getPackageName(), PackageManager.GET_META_DATA).metaData;
+			logger.log("ROHAN Kochava meta " + meta.get("kochavaAppGUID"));
```
Comment (jishnu7) [inline android/KochavaPlugin.java]: Hello Rohan Kochava!

File: android/KochavaPlugin.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import android.content.Context;
+import android.util.Log;
+import android.os.Bundle;
+import android.content.pm.ActivityInfo;
+import android.content.pm.PackageManager;
+import android.content.pm.PackageManager.NameNotFoundException;
+
+public class KochavaPlugin implements IPlugin {
+
+	private Feature kTracker;
```
Comment (rampr) [inline android/KochavaPlugin.java]: why are we using tabs or 4 spaces ?

File: android/KochavaPlugin.java (hunk 1 above)
Comment (rohanhc) [reply android/KochavaPlugin.java]: Removing...

File: android/KochavaPlugin.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+			String purchaseData = receiptdata.getString("purchaseData");
+			String dataSignature = receiptdata.getString("dataSignature");
+
+			logger.log("ROHAN Kochava purchaseData " + purchaseData);
+			logger.log("ROHAN Kochava dataSignature " + dataSignature);
+            // place the Purchase Data and Data Signature into an object to pass to the method
+            HashMap < String, String > receiptObject = new HashMap < String, String > ();
+            receiptObject.put("purchaseData", purchaseData);
+            receiptObject.put("dataSignature", dataSignature);
+            kTracker.eventWithReceipt("Purchase", json, receiptObject);
```
Comment (rampr) [inline android/KochavaPlugin.java]: indentation is wrong

File: android/KochavaPlugin.java (hunk 2 above)
Comment (rohanhc) [reply android/KochavaPlugin.java]: We use tabs in all .java files

File: android/KochavaPlugin.java (hunk 3 above)
Comment (rohanhc) [reply android/KochavaPlugin.java]: done

File: js/index.js [hunk 4]
```diff
@@ -20,25 +20,17 @@ var Kochava = Class(function () {
     });
   };
 
-  this.trackPurchase = function (receipt, item, revenue,
+  this.trackPurchase = function (uid, receipt, item, revenue,
```
Comment (rampr) [inline js/index.js:23]: Does iOS break because of this ?

File: android/KochavaPlugin.java (hunk 1 above)
Comment (rohanhc) [reply android/KochavaPlugin.java]: done

File: android/KochavaPlugin.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	public void setInstallReferrer(String referrer) {
+	}
+
+	public void onActivityResult(Integer request, Integer result, Intent data) {
+	}
+
+	public boolean consumeOnBackPressed() {
+		return true;
```
Comment (rampr) [inline android/KochavaPlugin.java]: why are we doing this ?

File: android/KochavaPlugin.java (hunk 2 above)
Comment (rampr) [reply android/KochavaPlugin.java]: We should have a single convention in the whole repo.

File: android/manifest.xml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+<uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
+<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+<!--END_PLUGINS_MANIFEST-->
+<!--START_PLUGINS_APPLICATION-->
+<!-- GameThrive -->
+<meta-data android:name="kochavaAppGUID" android:value="" />
+<receiver android:name ="com.kochava.android.tracker.ReferralCapture"
+android:exported ="true" >
+<intent-filter>
+<action android:name ="com.android.vending.INSTALL_REFERRER" />
```
Comment (rampr) [inline android/manifest.xml]: indentation issues

File: android/KochavaPlugin.java (hunk 5 above)
Comment (jishnu7) [reply android/KochavaPlugin.java]: needed to handle backbutton action

**Review by rampr — CHANGES_REQUESTED** [review]
Added comments against things that need to be changed.

File: android/KochavaPlugin.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+			String purchaseData = receiptdata.getString("purchaseData");
+			String dataSignature = receiptdata.getString("dataSignature");
+
+			// place the Purchase Data and Data Signature into an object to pass to the method
+			HashMap < String, String > receiptObject = new HashMap < String, String > ();
+			receiptObject.put("purchaseData", purchaseData);
+			receiptObject.put("dataSignature", dataSignature);
+			kTracker.eventWithReceipt("Purchase", json, receiptObject);
+		} catch (JSONException ex) {
```
Comment (jishnu7) [inline android/KochavaPlugin.java]: I don't think you need to pur the entire logic inside the try catch block.

File: android/KochavaPlugin.java [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+import android.content.pm.ActivityInfo;
+import android.content.pm.PackageManager;
+import android.content.pm.PackageManager.NameNotFoundException;
+
+public class KochavaPlugin implements IPlugin {
+
+	private Feature kTracker;
+	private Activity mActivity;
+	private Context mContext;
+	private boolean DEBUG = true;
```
Comment (jishnu7) [inline android/KochavaPlugin.java]: I thin you want this flag to be false by default, I guess.

File: android/manifest.xml (hunk 6 above)
Comment (jishnu7) [inline android/manifest.xml]: incorrect indent

File: android/KochavaPlugin.java (hunk 8 above)
Comment (rohanhc) [reply android/KochavaPlugin.java]: done

File: android/KochavaPlugin.java (hunk 7 above)
Comment (rohanhc) [reply android/KochavaPlugin.java]: done

File: android/manifest.xml (hunk 6 above)
Comment (rohanhc) [reply android/manifest.xml]: done

Comment (rohanhc) [reply android/manifest.xml]: done

File: js/index.js (hunk 4 above)
Comment (rohanhc) [reply js/index.js:23]: Doesn't look like it should

**Review by jishnu7 — APPROVED** [review]

File: android/KochavaPlugin.java (hunk 2 above)
Comment (rampr) [reply android/KochavaPlugin.java]: This is still there

**Comment by rampr** [comment]
indentation is still 4 spaces, the whole repo should have the same style

**Review by bijoshtj — APPROVED** [review]

**Review by rampr — APPROVED** [review]

**Pull request merged**